Gps and imu test scripts

DarthSethious1 opened the pull request.

DarthSethious1 committed:
New Test Scripts

Wrote test scripts for IMU and GPS, GPS needs work

DarthSethious1 committed:
IMU Fall Test

Had to get rid of all 'Serial.println' for fall test. Will need to bring those back for debugging GPS

DarthSethious1 committed:
Fall Test for IMU

Fall test for IMU, currently no version in Master

DarthSethious1 commented:
I should've been uploading bit by bit, sorry it's a large PR now

DarthSethious1 committed:
Headers

I added in the stuff that Cam wrote in his 'signal integration' branch to write headers to the CSV

DarthSethious1 committed:
Added Header Stuff that Actually Works

yup

CClarke513 committed:
Merge branch 'master' into GPS-and-IMU-Test-Scripts

CClarke513 committed:
Merge branch 'master' into GPS-and-IMU-Test-Scripts

CClarke513 reviewed (approved):
This looks good to me - I trust your work

bbrk24 commented on CAPS_INF.CSV:
I thought this file (the .CSV) wasn't supposed to be checked in?

CClarke513 replied:
I thought so too - I thought I added it to the .gitignore in master

bbrk24 commented on gps.h:
```diff
@@ -49,33 +49,48 @@ class GPS {
   } \
-  ++m_phase;
+  //Serial.print("Increasing m_phase..."); \
+  m_phase++; \
```
Edit to add: I want to be clear that this is the most serious concern out of all the comments I left.

A backslash at the end of a line means to act like it's all on one line, so this (`m_phase++`) is actually part of the comment!
The way macros work in C/C++, the body of the `#define` has to be all one line. I used backslashes to make it take up multiple lines to be easier to read, but the downside is that `//`-style comments can't go in the middle of it. Either delete the comment, or change it to the `/*...*/` style.

DarthSethious1 replied:
I will start deleting all 'Serial.println' when it comes to merging with master

bbrk24 commented on gps.h:
```diff
@@ -49,33 +49,48 @@ class GPS {
 
     switch (m_phase) {
     case 0:
+      //Serial.print("Has not started I2C comms, attempting now...");
```
I would either delete these entirely, or lock them behind some condition rather than commenting them out. Commenting out code is fine for local testing, but rarely belongs on master.

bbrk24 commented on test_GPS.ino:
```diff
@@ -0,0 +1,49 @@
+    delay(1000);
+
+  } else {
+    Serial.print("Unable to read values for some reason...");
```
Not gonna hold up the PR over this, but this one should probably be println

bbrk24 commented on test_IMU.ino:
```diff
@@ -0,0 +1,118 @@
+  //Serial.println("Connecting to sensor...");
+  testIMU.initialize();
+}
+/* THIS IS FOR STATIC TEST
```
Same as above about commented-out code

bbrk24 commented on test_IMU.ino:
```diff
@@ -0,0 +1,118 @@
+  fakeCoords.timestamp.milliseconds += (unsigned int)shortDelay;
+  if (int(fakeCoords.timestamp.milliseconds) >= 1000){
+    //Serial.println("Another Second Passed...");
+    fakeCoords.timestamp.seconds += (unsigned int)1;
```
You don't need to cast constants. If you really want to be sure, you can put the letter U after it (i.e. `1U`) to make it an unsigned constant, but that usually isn't necessary.

DarthSethious1 replied:
IDK how, but I didn't see this the first time. It was giving me trouble when I didn't cast it to the appropriate type. When I cast it like this it stopped giving me trouble.

bbrk24 commented on test_IMU.ino:
```diff
@@ -0,0 +1,118 @@
+  delayMicroseconds((shortDelay * 1000) - (micros() % (shortDelay * 1000))); 
+
+  //This is to make the time a lot easier for us to handle
+  fakeCoords.timestamp.milliseconds += (unsigned int)shortDelay;
```
`shortDelay` should probably be unsigned anyways.

bbrk24 reviewed (changes requested):
First time reviewing a PR on mobile so I might have missed something or some comments might show up in the wrong place. Hopefully what I mean is clear enough.

DarthSethious1 committed:
Corrections to PR

Corrected what Riley pointed out. no commented out 'serial.println' and some other minor things.

DarthSethious1 committed:
remove capsinf

bbrk24 commented on test_IMU.ino:
```diff
@@ -97,17 +60,14 @@ void loop(){
   //This is to make the time a lot easier for us to handle
-  fakeCoords.timestamp.milliseconds += (unsigned int)shortDelay;
+  fakeCoords.timestamp.milliseconds += shortDelay;
   if (int(fakeCoords.timestamp.milliseconds) >= 1000){
```
I missed this the first time around, but I believe this could instead be written
```cpp
if (fakeCoords.timestamp.milliseconds >= 1000U) {
```

DarthSethious1 committed:
Syntax Fix

1U instead of (unsigned int)1
